core/controller/src/main/scala/whisk/core/loadBalancer/InvokerSupervision.scala

```diff
@@ -150,14 +274,95 @@ class InvokerActor extends FSM[InvokerState, InvokerInfo] {
+        case _ -> UnHealthy => transid.mark(this, LoggingMarkers.LOADBALANCER_INVOKER_UNHEALTHY, s"$name is unhealthy", akka.event.Logging.WarningLevel)
+        case _ -> Healthy   => logging.info(this, s"$name is healthy")
+    }
+
+    /** Scheduler to send testActivations, if the invoker is UnHealthy */
     onTransition {
-        case Healthy -> Offline => logging.warn(this, s"$name is down")
-        case Offline -> Healthy => logging.info(this, s"$name is online")
+        case _ -> UnHealthy => {
+            invokeTestActions()
+            setTimer(InvokerActor.timerName, Tick, 1.minute, true)
+        }
+        case UnHealthy -> _ => cancelTimer(InvokerActor.timerName)
     }
 
     initialize()
+
+    /**
+     * Handling for active acks. This method saves the result (successful or unsuccessful)
+     * into an RingBuffer and checks, if the InvokerActor has to be changed to UnHealthy.
+     *
+     * @param wasActivationSuccessful: result of Activation
+     * @param buffer to be used
+     */
+    private def handleCompletionMessage(wasActivationSuccessful: Boolean, buffer: RingBuffer[Boolean]) = {
+        buffer.add(wasActivationSuccessful)
+
+        // If the current state is UnHealthy, this means, that the activeAck is from an testaction.
+        // If this is successful it seems like the Invoker is Healthy again. So we execute immediately
+        // a new testaction to remove the errors out of the RingBuffer as fast as possible.
+        if (wasActivationSuccessful && stateName == UnHealthy) {
+            invokeTestActions()
+        }
+
+        if (stateName == Healthy && wasActivationSuccessful) {
+            stay
+        } else {
+            // Goto UnHealthy if there are more errors than accepted in buffer, else goto Healthy
+            if (buffer.toList.count(_ == true) >= InvokerActor.bufferSize - InvokerActor.bufferErrorTolerance) {
+                gotoIfNotThere(Healthy)
+            } else {
+                gotoIfNotThere(UnHealthy)
+            }
+        }
+    }
+
+    /**
+     * Creates an Activationmessage with the given action and sends it to the InvokerPool.
+     * The InvokerPool redirects it to the invoker which is represented by this InvokerActor.
+     */
+    private def invokeTestActions() = {
+        InvokerPool.testAction.map { action =>
+            val activationMessage = ActivationMessage(
+                // Use the sid of the InvokerSupervisor as tid
+                transid = transid,
+                action = action.fullyQualifiedName(true),
+                // Use empty DocRevision to force the invoker to pull the action from db all the time
+                revision = DocRevision(),
+                // Authentication is not needed anymore at this point.
+                user = Identity(Subject("unhealthyInvokerCheck"), EntityName("unhealthyInvokerCheck"), AuthKey(UUID(), Secret()), Set[Privilege]()),
```

Comment:
if the Identity is not semantically important, should we have just one created for all requests?